From: Joel DeJesus <dejesus.joel@e-flamingo.jp>
Date: Tue, 11 Apr 2017 15:05:52 +0900
Subject: fixing soname problem

---
 lib/Makefile.am | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index 392b207..f5a5172 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -41,5 +41,8 @@ libccoin_la_SOURCES=	\
 	util.c		\
 	utxo.c		\
 	wallet.c
-# is the soname version info going to just be libccoin.so.0.0.0
-libccoin_la_LDFLAGS = -module -shared  -version-info 1:1:0
+# current:revision:age 
+libccoin_la_LDFLAGS = -module -shared  -version-info 0:1:0
+
+
+
